spec/requests/users_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :request do
+  describe 'GET /index' do
+    before(:example) { get '/users' }
+
+    it 'returns correct response status' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'renders correct template' do
+      expect(response).to render_template(:index)
+    end
+
+    it 'returns correct response body' do
+      expect(response.body).to include('Users List')
+    end
+  end
+end
```

Comment:
- [ ] @benardop great work on your tests kindly add tests for the budget controller and views as done in the group and users controller